File: frontend/src/types/cve.ts
import { Cpe } from './cpe';
export interface Cve {
  id: string;
  name: string | null;
  description: string | null;
  modifiedAt: Date;
  publishedAt: Date;
  status: string | null;
  cvssV2Source: string | null;
  cvssV2Type: string | null;
  cvssV2Version: string | null;
  cvssV2VectorString: string | null;
  cvssV2BaseScore: string | null;
  cvssV2BaseSeverity: string | null;
  cvssV2ExploitabilityScore: string | null;
  cvssV2ImpactScore: string | null;
  cvssV3Source: string | null;
  cvssV3Type: string | null;
  cvssV3Version: string | null;
  cvssV3VectorString: string | null;
  cvssV3BaseScore: string | null;
  cvssV3BaseSeverity: string | null;
  cvssV3ExploitabilityScore: string | null;
  cvssV3ImpactScore: string | null;
  cvssV4Source: string | null;
  cvssV4Type: string | null;
  cvssV4Version: string | null;
  cvssV4VectorString: string | null;
  cvssV4BaseScore: string | null;
  cvssV4BaseSeverity: string | null;
  cvssV4ExploitabilityScore: string | null;
  cvssV4ImpactScore: string | null;
  references: string[] | null;
  weaknesses: string[] | null;
  cpes: Cpe[];
}

File: backend/src/xfd_django/xfd_api/schema_models/cve.py
"""Cve schema."""
# Third-Party Libraries
# from pydantic.types import UUID1, UUID
# Standard Python Libraries
from datetime import datetime
from typing import Optional
from uuid import UUID

# Third-Party Libraries
from pydantic import BaseModel


class Cve(BaseModel):
    """Cve schema."""

    id: UUID
    name: Optional[str]
    published_at: datetime
    modified_at: datetime
    status: str
    description: Optional[str]
    cvss_v2_source: Optional[str]
    cvss_v2_type: Optional[str]
    cvss_v2_vector_string: Optional[str]
    cvss_v2_base_severity: Optional[str]
    cvss_v2_exploitability_score: Optional[str]
    cvss_v2_impact_score: Optional[str]
    cvss_v3_source: Optional[str]
    cvss_v3_type: Optional[str]
    cvss_v3_vector_string: Optional[str]
    cvss_v3_base_severity: Optional[str]
    cvss_v3_exploitability_score: Optional[str]
    cvss_v3_impact_score: Optional[str]
    cvss_v4_source: Optional[str]
    cvss_v4_type: Optional[str]
    cvss_v4_vector_string: Optional[str]
    cvss_v4_base_severity: Optional[str]
    cvss_v4_exploitability_score: Optional[str]
    cvss_v4_impact_score: Optional[str]
    weaknesses: Optional[str]
    references: Optional[str]

    class Config:
        """Config."""

        from_attributes = True
